### PR TITLE
Fixes #23585: Port the index and database parsing of rudder-package in rust

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23585

### CODE REVIEW

File: relay/sources/rudder-pkg/rust/rudder-package/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
+use std::{
+    collections::HashMap,
+    fs,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+struct PluginMetadata {
+    #[serde(rename(serialize = "type", deserialize = "type"))]
```
Comment (amousset) [inline relay/sources/rudder-pkg/rust/rudder-package/src/main.rs:10]:
```suggestion
    #[serde(rename = "type")]
```

and the same for others

File: relay/sources/rudder-pkg/rust/rudder-package/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    let data = fs::read_to_string("./tests/plugin_database_parsing.json").expect("Unable to parse file './tests/pluginèdatabase_parsing.json'");
+    let db: InstalledPluginDatabase = serde_json::from_str(&data).unwrap();
+    assert_eq!(db.plugins["rudder-plugin-aix"].metadata.plugin_type, "plugin");
+  }
+
+  #[test]
+  fn test_plugin_index_parsing() {
+    let data = fs::read_to_string("./tests/repo_index.json").expect("Unable to parse file './tests/repo_index.json'");
+    let index: RepoIndex = serde_json::from_str(&data).unwrap();
+    assert_eq!(index.0[0].metadata.version, "8.0.0~beta2-2.1");
```
Comment (amousset) [inline relay/sources/rudder-pkg/rust/rudder-package/src/main.rs:74]: You could defined the reference RepoIndex here for comparison (a bit verbose but doable)

File: relay/sources/rudder-pkg/rust/rudder-package/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    #[serde(rename(serialize = "jar-files", deserialize = "jar-files"))]
+    jar_files: Option<Vec<String>>,
+
+    #[serde(flatten)]
+    metadata: PluginMetadata,
+}
+
+#[derive(Serialize, Deserialize)]
+struct InstalledPluginDependency {
+    python: Option<Vec<String>>,
```
Comment (amousset) [inline relay/sources/rudder-pkg/rust/rudder-package/src/main.rs:50]: I think I'd go for a simpler Vec<String> here

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR updated with a new commit

**Pull request merged**